### PR TITLE
Added Game control functionalities.

### PR DESCRIPTION
**Pull request by kaushikb11**
Features for the Single Player Game mode to Start, Pause, Resume and Restart the game.
Peek 2019-03-29 18-26

### CODE REVIEW

File: client/src/components/Layout/GamePage.jsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-nested-ternary */
```
Comment (shivangg) [inline client/src/components/Layout/GamePage.jsx]: Don't disable this eslint rule. This is important. Nested ternary is making it difficult to understand code.

File: client/src/components/Layout/GamePage.jsx [hunk 2]
```diff
@@ -26,7 +27,9 @@ const GamePage = (props) => {
           <Grid item xs={3} />
           <Grid item xs={6}>
             <Button variant="outlined" size="medium" color="primary" onClick={startGame}>
-              Start
+              {status === 0 ? 'Start'
+                : status === 1 ? 'Pause'
+                  : 'Restart'}
```
Comment (shivangg) [inline client/src/components/Layout/GamePage.jsx]: Just use an if-else here.

File: client/src/components/PacmanGame/PacmanGame.jsx [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable import/no-unresolved */
```
Comment (shivangg) [inline client/src/components/PacmanGame/PacmanGame.jsx]: Why disabling this rule. Just yarn under the `client` folder to resolve this error.

File: client/src/components/PacmanGame/PacmanGame.jsx [hunk 4]
```diff
@@ -62,7 +63,24 @@ class PacmanGame extends Component {
...
+    if (status === 2) {
+      this.setInitialGameState();
+      this.setState({
+        score: 0,
+        moveGhostsCount: 0,
+      });
+      this.setState({ status: 1 });
+    }
+    if (status === 0) this.setState({ status: 1 });
+    joinGame(pacman);
```
Comment (shivangg) [inline client/src/components/PacmanGame/PacmanGame.jsx:82]: Difficult to remember what status code is for what situation. Use a function or object instead.

**Review by shivangg — CHANGES_REQUESTED** [review]

File: client/src/components/PacmanGame/PacmanGame.jsx (hunk 4 above)
Comment (kaushikb11) [reply client/src/components/PacmanGame/PacmanGame.jsx:82]: There's a comment in the State object of the PacmanGame component

Comment (shivangg) [reply client/src/components/PacmanGame/PacmanGame.jsx]: Yeah, that comment is non-authoritative. Replace that with a function or object.

File: client/src/components/Layout/GamePage.jsx (hunk 1 above)
Comment (shivangg) [reply client/src/components/Layout/GamePage.jsx]: @kaushikb11 Why didn't you remove it?

File: client/src/components/PacmanGame/PacmanGame.jsx (hunk 4 above)
Comment (shivangg) [reply client/src/components/PacmanGame/PacmanGame.jsx:82]: Yeah, that comment is non-authoritative. Replace that with a function or object.

**Review by shivangg — CHANGES_REQUESTED** [review]
Remove the Merge commit and replace by Rebase

**Pull request closed without merging**